linkify only on post details page

aditilamkhede opened the pull request:
### What does this pull request aim to achieve? 💯
convert text to url only when on post details page. If we are on helpborad(i.e. feed/posts) page, it should only be displayed as plain text.

_Please be concise and link any related issue(s):_ Resolves issue #2153 

<!--### 🚨Before submitting this pull request🚨:-->

<!--_Please do **NOT** submit this PR if you have not done the following:_-->

- [x] I have checked that no one else is working on similar changes.
- [ ] I have read the latest [**README**](README.md) and understand the development workflow.
- [x] The name of this branch is: **`feature/<branch_name>`**, or **`hotfix/<branch_name>`** if this is a hotfix, where `<branch_name>` briefly describes the issue(s) resolved and is prefixed with the issue number(s) (e.g. `1127-1130-update-git-branching-model`). The branch is created in a cloned version of this repo, **not a forked repo**.
- [x] This branch is rebased/merged with the **latest staging** (or the **latest production** if it is a hotfix).
- [x] There are no merge conflicts.
- [x] There are no console warnings and errors.
- [x] On the right hand side of this PR, I have linked the appropriate issue(s) under "Issues" and added a project progress under "Projects".
- [x] The title describes the issue(s) being addressed using format: **`<issue_numbers> - <brief_issue_description>`** (e.g. `#1127/#1130 - Update Git Branching Model`)

aditilamkhede committed:
linkify only on postdetails page

convert text to url only when on post details page. If we are on helpborad(i.e. feed/posts) page, it should only be displayed as plain text.

ddveloper commented on client/src/components/Feed/Post.js:
```diff
@@ -842,18 +843,26 @@ const Post = ({
       <Heading level={4} className="h4">
-        <Highlight textObj={linkify(title)} highlight={highlightWords} />
+      {
+        linkifyText ? <Highlight textObj={linkify(title)} highlight={highlightWords} /> :
```
Hi @aditilamkhede , could this be simplified as `<Highlight textObj={linkifyText ? linkify(title) : title} highlight={highlightWords} />` ?

ddveloper commented on client/src/components/Feed/Post.js:
```diff
@@ -842,18 +843,26 @@ const Post = ({
       <p className="post-description">
-        <Highlight textObj={linkify(finalContent)} highlight={highlightWords} />
+        {linkifyText ?
+          <Highlight textObj={linkify(finalContent)} highlight={highlightWords} />
```
same here

aditilamkhede replied:
Yes, makes sense. will do that.

aditilamkhede committed:
simplified linkify coniditions in post.js

ddveloper reviewed (dismissed):
LGTM! thanks @aditilamkhede

rabbi08 reviewed (approved):
Verified the fix as working fine, the link added in the post are displayed as link in the post details page. On the help board, it is displayed as text. Thanks.

Please refer attached video:

https://user-images.githubusercontent.com/67168400/104820700-ac5fb900-5804-11eb-871c-a45c928a6901.mov

ddveloper commented:
Hi @aditilamkhede , I found 2 other places need set `linkify` to false, they are in `OrgPost.js` and `Activity.js`, maybe we can give a default `false` for the input `linkify` and only set it to be `true` in `PostPage.js`.

Besides, `linkify` itself sounds not straightforward, how about a name like `needLinkifyText`?

aditilamkhede commented:
> Hi @aditilamkhede , I found 2 other places need set `linkify` to false, they are in `OrgPost.js` and `Activity.js`, maybe we can give a default `false` for the input `linkify` and only set it to be `true` in `PostPage.js`.
> 
> Besides, `linkify` itself sounds not straightforward, how about a name like `needLinkifyText`?

@ddvelope yes, Thanks will make these changes as well. I was also thinking there might be more places where POST component is being used. 

Name of the boolean variable is "linkifyText", I couldn't think of better name, "islinkifyText" wasn't a good choice either nor did "doLinkifyText". 

we can call it "linkifyText", "needLinkifyText", "isURL", "textToURL", "convertTextToURL"? let me know which one fits better with the functionality?

ddveloper commented:
I think `convertTextToURL` is good 😃 , there won't be other `<Post>`, you can double confirm with a global search
![image](https://user-images.githubusercontent.com/44758313/104829227-9c58d100-5826-11eb-8d85-8cd1b4bbc846.png)

aditilamkhede commented:
> I think `convertTextToURL` is good 😃 , there won't be other `<Post>`, you can double confirm with a global search
> ![image](https://user-images.githubusercontent.com/44758313/104829227-9c58d100-5826-11eb-8d85-8cd1b4bbc846.png)
so the change the name of variable in POST component to "convertTextToURL", how about the function name linkify should we change that too?

ddveloper commented:
function name `linkify` is good, sounds like a verb 👍

rabbi08 reviewed (dismissed):
was looking for some verifications, only. : Done
Thanks.

aditilamkhede committed:
changed variable name to convertTextToURL

changed variable name to convertTextToURL and added variable to orgpost.js and Activity.js

aditilamkhede commented:
@ddveloper @rabbi08 Please review with changes suggested by @ddveloper.

ddveloper reviewed (approved):
LGTM

rabbi08 reviewed (approved):
Thanks @aditilamkhede 
Verified the fix on feature branch, for Offer/Request post for Individual and Org post. The link text in the post is displayed as link on post description page only else it is displayed as text. 

Refer below GIF
![ezgif com-video-to-gif-10](https://user-images.githubusercontent.com/67168400/104861265-24a9a580-58fd-11eb-8dbd-803e6e3bf3ee.gif)

Zaydme reviewed (approved):
LGTM
Thank you @aditilamkhede
